docs/src/main/asciidoc/extension-maturity-model.adoc

```diff
@@ -0,0 +1,93 @@
+
+[id="extension-maturity-model"]
+= A maturity model for Quarkus extensions
+include::_attributes.adoc[]
+:diataxis-type: concept
+:categories: writing-extensions
+:topics: extensions
+:summary: Quarkus extensions can do a lot, or a little. This guide explains some of the capabilities extension authors may wish to include.
+////
+The document header ends at the first blank line. Do not remove the blank line between the header and the abstract summary.
+////
+
+What makes a good Quarkus extension? What capabilities is a Quarkus extension expected to provide?
+
+* It actually works
+** Works in JVM mode
+** Works in dev mode
+** Works in native
+* Dev service (if there is an external service dependency)
+* Developer joy
+** Configuration support
+** Use build-time application knowledge to remove boilerplate
+* Dev UI
+* Supersonic subatomic performance
+** Use build-time application knowledge to eliminate wasteful runtime code paths
+* Codestart application template
```

Comment:
```suggestion
* Runtime actually works
** Works in JVM mode
** Works in native
* Supersonic subatomic performance
** Use build-time application knowledge to eliminate wasteful runtime code paths
* Developer Joy
 ** Configuration support
 ** Dev service (if there is an external service dependency)
 ** Use build-time application knowledge to remove boilerplate
 ** Dev UI
 ** Codestart application template
```

Totally optional suggestion on grouping it a little bit different as the existing one (at least to me) felt as its categories overlapped a bit too much.

Reply:
I think if we group too much under developer joy it starts to break the ranking. I'd say that tolerating dev mode is table stakes for an extension, but 'redefine the programming model' is a pretty advanced feature. 

What about pulling developer joy some sort of tag or 'key'. So Config and Boilerplate reduction become top level items, dev mode goes back to being in front of native mode, and we put the joy icon at the beginning or end of those lines. Or if we do boxes, the joy icon tags the box.

Then we could do a similar tagging for the others, with some sort of 'speed' icon (I don't think we actually have one, but we should?). So then people can glance at the list and see how each thing fits in. Something like this, but not looking terrible:

<img width="680" alt="image" src="https://github.com/user-attachments/assets/1d93a205-2b88-4412-8095-716e87f6181d">

Also, in my crappy mock-up I didn't bother to adjust the order of items, and I had to steal an icon since we don't have a speed one.

Reply:
Or something like this, if we went for a more graphical maturity model representation:
<img width="1274" alt="image" src="https://github.com/user-attachments/assets/49fd261f-342f-4b4f-a0bd-9203bea746bb">

(Again, wrong icon for performance, and I only filled out the first bit.)

Reply:
I've changed 'it' to 'runtime', as per your suggestion. I've (hopefully) fixed the problem with Developer Joy you pointed out by just removing all of the 'developer joy' nesting. I've not done anything yet with icons or moving to a graphical format.